utils/doclint/generateDotnetApi.js

```diff
@@ -314,19 +314,37 @@ function generateNameDefault(member, name, t, parent) {
   let enumName = generateEnumNameIfApplicable(member, name, t, parent);
   if (!enumName && member) {
     if (member.kind === 'method' || member.kind === 'property') {
-      // this should be easy to name... let's call it the same as the argument (eternal optimist)
-      let probableName = `${parent.name}${translateMemberName(``, name, null)}`;
-      let probableType = additionalTypes.get(probableName);
-      if (probableType) {
-        // compare it with what?
-        if (probableType.expression != t.expression) {
-          throw new Error(`Non-matching types with the same name. Panic.`);
+      let names = [
+        parent.name,
+        translateMemberName(``, member.name, null),
+        translateMemberName(``, name, null),
+      ];
+      if (names[2] === names[1])
+        names.pop(); // get rid of duplicates, cheaply
+      let attemptedName = names.pop();
+      let compareType = function (left, right) {
```

Comment:
Can you change it into isSameType or something, otherwise it seems that the method returns true if the types are different  but I don't quite understand how `if (!left || !right) return false;` statement fits into that logic.